### PR TITLE
Update ZK client version to 3.9.1

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs

### CODE REVIEW

**Review by baldersheim — DISMISSED** [review]

File: vespa-dependencies-enforcer/allowed-maven-dependencies.txt [hunk 1]
```diff
@@ -128,9 +128,11 @@ org.apache.opennlp:opennlp-tools:${opennlp.vespa.version}
 org.apache.velocity:velocity-engine-core:${velocity.vespa.version}
 org.apache.yetus:audience-annotations:0.12.0
 org.apache.zookeeper:zookeeper-jute:${zookeeper.client.vespa.version}
+org.apache.zookeeper:zookeeper-jute:3.8.0
 org.apache.zookeeper:zookeeper-jute:3.8.1
 org.apache.zookeeper:zookeeper-jute:3.9.1
 org.apache.zookeeper:zookeeper:${zookeeper.client.vespa.version}
+org.apache.zookeeper:zookeeper:3.8.0
```
Comment (bjorncs) [inline vespa-dependencies-enforcer/allowed-maven-dependencies.txt:135]: This looks unexpected. Looks like we still have 3.8.0 being included somewhere else.

Comment (jonmv) [reply vespa-dependencies-enforcer/allowed-maven-dependencies.txt:135]: `vespa-zookeeper-server` uses that version. All those submodules end up in here.

Comment (jonmv) [reply vespa-dependencies-enforcer/allowed-maven-dependencies.txt:135]: 3.8.1 is not used anywhere at all, except for in the 3.8.1 module, as far as I can see? And I intend to remove those modules when 3.9.1. is completely rolled out.

**Comment by jonmv** [comment]
But I need to address
```
[12:02:35.450] [2023-12-05 12:02:34.895] WARNING -                .org.apache.zookeeper.server.NettyServerCnxnFactory	Exception caught\nexception=\nio.netty.handler.codec.DecoderException: javax.net.ssl.SSLHandshakeException: Received fatal alert: certificate_unknown
``` before we can merge this. Hang on.

File: vespa-dependencies-enforcer/allowed-maven-dependencies.txt (hunk 1 above)
Comment (bjorncs) [reply vespa-dependencies-enforcer/allowed-maven-dependencies.txt:135]: The `zookeeper.version` property should be used then.

**Comment by jonmv** [comment]
Superseded by #29658

**Pull request closed without merging**